### PR TITLE
Provide configuration options for Github Enterprise

### LINKED ISSUE #30: Github Enterprise Support
* ~~modify passport module to support github enterprise~~
* provide the necessary configuration options
* use the original [jaredhanson/passport-github](https://github.com/jaredhanson/passport-github) with Github Enterprise support

### PR DESCRIPTION
**Pull request by mfellner**
Fixes #30

### CODE REVIEW

File: config.yml [hunk 1]
```diff
@@ -14,7 +14,10 @@ GITHUB_SCOPES:
   - user:email
   - public_repo
   - admin:repo_hook
-GITHUB_URL: https://api.github.com
+# See https://developer.github.com/v3/enterprise
+# for GitHub Enterprise configuration.
+GITHUB_MAIN_ADDR: https://github.com
+GITHUB_API_ADDR: https://api.github.com
```
Comment (zappr-test) [inline config.yml]: `GITHUB_API_URL`, `GITHUB_UI_URL`?

File: server/passport.js [hunk 2]
```diff
@@ -24,6 +26,15 @@ function normalizeProfile(profile) {
 }
 
 /**
+ * @param {string} root - http(s)://domain
+ * @param {string} path - /some/path
+ * @returns {string} - http(s)://domain/some/path
+ */
+function urlJoin(root, path) {
```
Comment (prayerslayer) [inline server/passport.js]: `joinURL`?

File: server/service/GithubService.js [hunk 3]
```diff
@@ -23,14 +25,12 @@ const PATHS = {
...
-    let url = nconf.get('GITHUB_URL') + path
-
     return {
       json: true,
       method: method,
-      url,
+      url: GITHUB_API_ADDR + path,
       headers: {
-        'User-Agent': 'ZAPPR/1.0 (+https://zappr.hackweek.zalan.do)',
+        'User-Agent': `ZAPPR/1.0 (+${HOST_ADDR})`,
```
Comment (prayerslayer) [inline server/service/GithubService.js]: I'd use either just `Zappr (+address)` or a real version, like the first 8 chars of the commit id.

**Comment by prayerslayer** [comment]
I think there might be some uses of the `GITHUB_URL` nconf variable that you didn't replace?

File: config.yml (hunk 1 above)
Comment (mfellner) [inline config.yml]: UI_URL sounds a bit weird, I think. Passport uses that url for the [management console](https://developer.github.com/v3/enterprise/management_console/).

File: server/service/GithubService.js (hunk 3 above)
Comment (mfellner) [inline server/service/GithubService.js]: Ok. Turns out the version is [optional](https://tools.ietf.org/html/rfc7231#section-5.5.3) so I'm going to use the first suggestion.

File: server/service/GithubService.js [hunk 4]
```diff
@@ -225,6 +224,7 @@ export default class GithubService {
       const pages = await Promise.all(pageDefs.map(async (page) => await that.fetchRepoPage(page, accessToken)))
       pages.forEach(p => Array.prototype.push.apply(repos, p.body))
     }
+    info('Loaded %d repos from Github', repos.length)
```
Comment (prayerslayer) [inline server/service/GithubService.js]: Would log this as debug as it will be logged vor every user syncing github repos with upstream.

**Comment by prayerslayer** [comment]
Aside from merge conflicts and the one log line it's fine for me :+1:

**Comment by prayerslayer** [comment]
:+1:

**Pull request merged**